lib/Transforms/IPO/StructFieldCacheAnalysis.cpp

```diff
@@ -539,6 +71,48 @@ StructFieldAccessManager::getFieldAccessOnInstruction(
   return ret;
 }
 
+void StructFieldAccessManager::summarizeFunctionCalls() {
+  for (auto &it : StructFieldAccessInfoMap) {
+    it.second->summarizeFunctionCalls();
+  }
+}
+
+void StructFieldAccessManager::applyFiltersToStructs() {
+  // TODO: This function needs more work to add more filters to reduce the
+  // number of structs for analysis
+  DEBUG_WITH_TYPE(DEBUG_TYPE_IR, dbgs() << "To apply filters to structs\n");
+  for (auto it = StructFieldAccessInfoMap.begin();
+       it != StructFieldAccessInfoMap.end();) {
+    if (!it->second->isEligible()) {
+      delete it->second;
+      auto ToRemove = it++;
+      StructFieldAccessInfoMap.erase(ToRemove);
+      addStats(DebugStats::DS_PassedIntoOutsideFunction);
+    } else if (it->second->getTotalNumFieldAccess() <
+               MinimalAccessCountForAnalysis) {
+      delete it->second;
+      auto ToRemove = it++;
+      StructFieldAccessInfoMap.erase(ToRemove);
+      addStats(DebugStats::DS_NoAccess);
+    } else {
+      HotnessAnalyzer->addStruct(it->second);
+      it++;
+    }
+  }
+  DEBUG_WITH_TYPE(DEBUG_TYPE_STATS, HotnessAnalyzer->generateHistogram());
+  for (auto it = StructFieldAccessInfoMap.begin();
+       it != StructFieldAccessInfoMap.end();) {
+    if (!HotnessAnalyzer->isHot(it->second)) {
```

Comment:
The first loop needs to go through all the structs in the program and obtains their max hotness. Then the second loop comes in the filter out some structs if there are colder than a ratio of max hotness.